src/pages/index.tsx

```diff
@@ -12,40 +13,45 @@ type Props = {
 }
 
 const AllCommunities = ({ communities }: Props): JSX.Element => {
-  const { account, library } = useWeb3React()
-
-  const isConnected = typeof account === "string" && !!library
+  const refYours = useRef<HTMLDivElement>(null)
+  const refAccess = useRef<HTMLDivElement>(null)
+  const refOther = useRef<HTMLDivElement>(null)
 
   return (
     <Layout
       title="All communities on Agora"
       bg="linear-gradient(white 0px, var(--chakra-colors-gray-100) 700px)"
     >
-      <Stack spacing={8}>
-        <CategorySection title="Your communities">
-          {isConnected ? (
-            <Text>You're not part of any communities yet</Text>
-          ) : (
-            <div>Wallet not connected</div>
-          )}
-        </CategorySection>
-
-        <CategorySection title="Communities you have access to">
-          {isConnected ? (
-            <Text>You don't have access to any communities</Text>
-          ) : (
-            <div>Wallet not connected</div>
-          )}
-        </CategorySection>
-
-        <CategorySection title="All communities">
-          <SimpleGrid columns={{ base: 1, lg: 2 }} spacing={10}>
-            {communities.map((i) => (
-              <CommunityCard community={i} key={i.id} />
-            ))}
-          </SimpleGrid>
-        </CategorySection>
-      </Stack>
+      <>
+        <Stack spacing={8}>
+          <CategorySection
+            title="Your communities"
+            placeholder={"You're not part of any communities yet"}
+            ref={refYours}
+          />
+          <CategorySection
+            title="Communities you have access to"
+            placeholder={"You don't have access to any communities"}
+            ref={refAccess}
+          />
+          <CategorySection
+            title="Other communities"
+            placeholder={"There aren't any other communities"}
+            ref={refOther}
+          />
+        </Stack>
+        {communities.map((community) => (
+          <CommunityProvider data={community} key={community.id}>
```

Comment:
I think we should have a comment about why need to wrap it with the CommunityProvider too